### DISCUSSION TITLE
AU - Postcode according to State

### DISCUSSION THREAD

**Discussion by casporo** [faker-js/faker#2647] [Ideas]
Hi Devs,

I am an Automation QA working on a project, and we have implemented Faker JS. We implemented AU localisation though when we tried to extract postcode based on state, we could not map it to the state as our endpoint will ensure if for example the state is New South Wales, the postcode must start with 2XXX. 

This is the function that we implemented: `faker.address.zipCode(state: 'NSW')`. If we do this it mentioned localisation not supported. 

We followed the following documentation stated [here](https://v6.fakerjs.dev/api/address.html)

**Comment by ST-DDT** [comment]
The en_AU locale does not have the required mapping table here:
https://github.com/faker-js/faker/tree/next/src/locales/en_AU/location
See here for an example:
https://github.com/faker-js/faker/blob/next/src/locales/en_US/location/postcode_by_state.ts
You can provide that at runtime via:
`faker.rawDefinitions.location.postcode_by_state = {...}`
Please consider creating a PR to contribute the data to the project.

**Reply by casporo** [reply]
Thanks @ST-DDT, I can consider creating a PR to contribute the data for this project.

**Reply by matthewmayer** [reply]
https://en.wikipedia.org/wiki/Postcodes_in_Australia#Allocation

One problem i see is that there are multiple non-consecutive ranges of postcodes, for example 2000—2599 2619—2899 are NSW, but 2600—2618 is ACT.

We might need to modify the postcode_by_state definitions to allow for multiple fake patterns per state.

**Reply by matthewmayer** [reply]
Passing an array of fake patterns is supported at runtime, but typescript doesn't like this at compile time:

```js
faker.rawDefinitions.location.postcode_by_state = {
    "NSW": ['{{number.int({"min":2000, "max":2599})}}', '{{number.int({"min":2619, "max":2899})}}'],
    "ACT": ['{{number.int({"min":2600, "max":2618})}}', '{{number.int({"min":2900, "max":2920})}}']
}
```

**Comment by ProxiBlue** [comment]
Hello @casporo 

Did you ever get round to doing a PR with the information mate ?
I could not see/find one.

If not, did you work out postcode for all states and can you share your config.

**Reply by matthewmayer** [reply]
The good news is that the problem I mentioned here as now resolved as we do allow multiple fake patterns per state

https://github.com/faker-js/faker/discussions/2647#discussioncomment-8403241

**Comment by ProxiBlue** [comment]
AI generated :)

```
faker.rawDefinitions.location.postcode_by_state = {
    NSW: [
        '{{number.int({"min": 2000,"max": 2599})}}',
        '{{number.int({"min": 2619,"max": 2899})}}'
    ],
    ACT: [
        '{{number.int({"min": 2600,"max": 2618})}}',
        '{{number.int({"min": 2900,"max": 2920})}}'
    ],
    VIC: [
        '{{number.int({"min": 3000,"max": 3999})}}',
        '{{number.int({"min": 8000,"max": 8999})}}'
    ],
    QLD: [
        '{{number.int({"min": 4000,"max": 4999})}}',
        '{{number.int({"min": 9000,"max": 9999})}}'
    ],
    SA: [
        '{{number.int({"min": 5000,"max": 5799})}}',
        '{{number.int({"min": 5800,"max": 5999})}}'
    ],
    WA: [
        '{{number.int({"min": 6000,"max": 6797})}}',
        '{{number.int({"min": 6800,"max": 6999})}}'
    ],
    TAS: [
        '{{number.int({"min": 7000,"max": 7799})}}',
        '{{number.int({"min": 7800,"max": 7999})}}'
    ],
    NT: [
        '{{number.int({"min": 0800,"max": 0899})}}',
        '{{number.int({"min": 0900,"max": 0999})}}'
    ]
}
```

**Comment by ProxiBlue** [comment]
Created a PR: https://github.com/faker-js/faker/pull/3219
